sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -86,10 +96,24 @@ case class GenerateExec(
     child.execute().mapPartitionsWithIndexInternal { (index, iter) =>
       val generatorNullRow = new GenericInternalRow(generator.elementSchema.length)
       val rows = if (join) {
+
+        lazy val project = UnsafeProjection.create(
+          projectedChildOutput,
+          child.output,
+          subexpressionEliminationEnabled)
```

Comment:
no need to set this. It's a very simple projection that only deal with attributes

Reply:
we can simplify the code and centralize all the logic here
```
val pruneChildForResult: InternalRow => InternalRow = if (omitGeneratorReferences) {
  val generatorReferences = generator.references
  val requiredChildOutput = child.output.filterNot(generatorReferences.contains)
  UnsafeProjection.create(requiredChildOutput, child.output)
} else {
  identity
}
```